src/components/Modals/NewMessageModal.jsx

```diff
@@ -0,0 +1,215 @@
+// React Imports
+import React, { useState, useContext } from 'react';
+// Inrupt Library Imports
+import { useSession } from '@hooks';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+// Utility Imports
+import { sendMessageTTL, getMessageTTL } from '../../utils';
+// Context Imports
+import { MessageContext, SignedInUserContext } from '../../contexts';
+
+/**
+ * messageFolderProps is an object that stores the props for the MessageFolder
+ * component
+ *
+ * @typedef newMessageModalProps
+ * @type {object}
+ * @property {() => void} closeForm - The function used to trigger NewMessageModal to
+ * close
+ * @memberof typedefs
+ */
+
+/**
+ * NewMessageModal Component - Component that allows user to write
+ *  a message to another user from their inbox
+ *
+ * @memberof Modals
+ * @name NewMessageModal
+ * @param {newMessageModalProps} Props - Props used for NewMessageModal
+ * @returns {React.JSX.Element} React component for NewMessageModal
+ */
+
+const NewMessageModal = ({ showModal, setShowModal, oldMessage = '' }) => {
+  const { session } = useSession();
+  const { outboxList, setOutboxList } = useContext(MessageContext);
+  const { podUrl } = useContext(SignedInUserContext);
+  const [originalMessage, setOriginalMessage] = useState(oldMessage.message);
+
+  const [message, setMessage] = useState({
+    recipientPodUrl: oldMessage ? oldMessage.senderWebId.split('profile')[0] : '',
+    title: oldMessage ? `RE: ${oldMessage.title}`.replace('RE:RE:', 'RE:') : '',
+    message: '',
+    inReplyTo: oldMessage ? oldMessage.messageId : '',
+    messageUrl: oldMessage ? oldMessage.messageUrl : ''
+  });
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState('');
+  const [successTimeout, setSuccessTimeout] = useState(false);
+
+  // Modifies message upon input
+  const handleChange = (e) => {
+    setMessage({
+      ...message,
+      [e.target.id]: e.target.value
+    });
+  };
+
+  // Handles submit (awaiting functionality for this)
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!message.title) {
```

Comment:
I think we can make these errors a little more user friendly/readable, dropping the "value for the".